src/code.cloudfoundry.org/cni-wrapper-plugin/netrules/masquerader_test.go

```diff
@@ -0,0 +1,316 @@
+						),
+						ghttp.CombineHandlers(
+							ghttp.VerifyRequest("GET", "/"),
+							ghttp.RespondWith(http.StatusInternalServerError, []byte{}),
+						),
+						ghttp.CombineHandlers(
+							ghttp.VerifyRequest("GET", "/"),
+							ghttp.RespondWith(http.StatusInternalServerError, []byte{}),
+						),
+						ghttp.CombineHandlers(
+							ghttp.VerifyRequest("GET", "/"),
+							ghttp.RespondWith(http.StatusOK, silkDaemonResponseBytes),
+						),
+					)
+				})
+
+				It("eventually succeeds", func() {
+					expectedRule := rules.IPTablesRule{"--source", containerIP, "!", "-o", vtepName, "!", "--destination", leaseCIDR, "--jump", "MASQUERADE"}
+					err := masquerader.AddIPMasq()
+					Expect(err).NotTo(HaveOccurred())
+					Expect(fakeIPTables.BulkAppendCallCount()).To(Equal(1))
+					table, chain, rules := fakeIPTables.BulkAppendArgsForCall(0)
+					Expect(table).To(Equal("nat"))
+					Expect(chain).To(Equal("POSTROUTING"))
+					Expect(rules).To(HaveLen(1))
+					Expect(rules[0]).To(Equal(expectedRule))
+				})
+			})
+		})
+
+		Context("when customNoMasqueradeCIDRRange is set", func() {
+			var customNoMasqueradeCIDRRange string
+
+			BeforeEach(func() {
+				customNoMasqueradeCIDRRange = "10.33.0.0/16"
+				masquerader.CustomNoMasqueradeCIDRRange = customNoMasqueradeCIDRRange
+			})
+
+			It("makes a masq rule with the customNoMasqueradeCIDRRange", func() {
+				expectedRule := rules.IPTablesRule{"--source", containerIP, "!", "-o", vtepName, "!", "--destination", customNoMasqueradeCIDRRange, "--jump", "MASQUERADE"}
+				err := masquerader.AddIPMasq()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(fakeIPTables.BulkAppendCallCount()).To(Equal(1))
+				table, chain, rules := fakeIPTables.BulkAppendArgsForCall(0)
+				Expect(table).To(Equal("nat"))
+				Expect(chain).To(Equal("POSTROUTING"))
+				Expect(rules).To(HaveLen(1))
+				Expect(rules[0]).To(Equal(expectedRule))
+			})
+
+			Context("when IPtables errors", func() {
+				BeforeEach(func() {
+					fakeIPTables.BulkAppendReturns(errors.New("meow-bad"))
+				})
+
+				It("returns an error", func() {
+					err := masquerader.AddIPMasq()
+					Expect(err).To(MatchError("meow-bad"))
+				})
+			})
+		})
+	})
+
+	Describe("DelIPMasq", func() {
+		Context("when customNoMasqueradeCIDRRange is NOT set", func() {
+			var fakeSilkDaemonServer *ghttp.Server
+			var silkDaemonResponseBytes []byte
+
+			BeforeEach(func() {
+				fakeSilkDaemonServer = ghttp.NewServer()
+				daemonPort := strings.Split(fakeSilkDaemonServer.Addr(), ":")[1]
+				masquerader.DaemonPort = daemonPort
+
+				silkDaemonResponse := daemon.NetworkInfo{
+					OverlaySubnet: leaseCIDR,
+				}
+
+				var err error
+				silkDaemonResponseBytes, err = json.Marshal(silkDaemonResponse)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				fakeSilkDaemonServer.Close()
+			})
+
+			Context("when the silk daemon responds with the lease", func() {
```

Comment:
Is this a new dependency? Tearing down a container/stopping an app now require the silk daemon to be available still?

Reply:
Good question!

**It is not a new dependency for container up.**
The other silk-cni requires it. So it is a new dependency for silk-cni-wrapper, but not for container up. 

**It is a new dependency for container down.**
But if it fails, it logs a failure and doesn't exit early (which is how we always handled container down issues).